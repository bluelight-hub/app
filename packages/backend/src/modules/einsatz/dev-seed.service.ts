import { Inject, Injectable, LoggerService, OnModuleInit } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { EinsatzService } from './einsatz.service';

/**
 * Erstellt im Entwicklungsmodus automatisch einen Einsatz, falls keiner existiert.
 */
@Injectable()
export class DevSeedService implements OnModuleInit {
    constructor(
        private readonly einsatzService: EinsatzService,
        private readonly configService: ConfigService,
        @Inject('Logger') private readonly logger: LoggerService,
    ) { }

    async onModuleInit() {
        if (this.configService.get<string>('NODE_ENV') !== 'development') {
            return;
        }
        if (this.configService.get<string>('SEED_INITIAL_EINSATZ') === 'false') {
            this.logger.debug('Initialer Einsatz-Seed deaktiviert');
            return;
        }
        const count = await this.einsatzService.count();
        if (count > 0) {
            return;
        }
        const name = this.configService.get<string>('DEV_EINSATZ_NAME')
            || `Dev-Einsatz ${new Date().toISOString()}`;
        await this.einsatzService.create({ name });
        this.logger.log(`Initialer Einsatz "${name}" wurde angelegt`);
    }
}
