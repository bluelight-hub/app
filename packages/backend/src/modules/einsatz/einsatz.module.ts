import { CommonModule } from '@/common/common.module';
import { PrismaModule } from '@/prisma/prisma.module';
import { Module } from '@nestjs/common';
import { EinsatzController } from './einsatz.controller';
import { EinsatzService } from './einsatz.service';

/**
 * Modul für die Einsatzverwaltung
 *
 * Dieses Modul implementiert die zentrale Geschäftslogik für
 * die Verwaltung von Einsätzen im Bluelight Hub System.
 * Es bietet Funktionalitäten zur Erstellung, Bearbeitung
 * und Verwaltung von Einsätzen.
 *
 * Features:
 * - CRUD-Operationen für Einsätze
 * - Einsatzstatusmanagement
 * - Integration mit anderen Modulen (ETB, Audit)
 * - Berechtigungsbasierte Zugriffskontrolle
 *
 * @module EinsatzModule
 */
@Module({
  imports: [PrismaModule, CommonModule],
  controllers: [EinsatzController],
  providers: [EinsatzService],
  exports: [EinsatzService],
})
export class EinsatzModule {}
