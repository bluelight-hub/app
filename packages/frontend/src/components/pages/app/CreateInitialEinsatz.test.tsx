import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import { useCreateEinsatz } from '../../../hooks/einsatz/useEinsatzQueries';
import CreateInitialEinsatz from './CreateInitialEinsatz';

// Mock der Hooks
vi.mock('../../../hooks/einsatz/useEinsatzQueries', () => ({
    useCreateEinsatz: vi.fn(),
    useEinsatzOperations: vi.fn(() => ({
        hasEinsatz: vi.fn(() => false),
        hasDataBeenFetched: true,
    })),
}));

// Mock für useNavigate
const mockNavigate = vi.fn();
vi.mock('react-router', () => ({
    ...vi.importActual('react-router'),
    useNavigate: () => mockNavigate,
}));

// Mock für Ant Design message
vi.mock('antd', async () => {
    const antd = await vi.importActual('antd');
    return {
        ...antd,
        message: {
            success: vi.fn(),
            error: vi.fn(),
        },
    };
});

describe('CreateInitialEinsatz', () => {
    // Mock-Objekte für die Tests
    const mockMutateAsync = vi.fn();
    const mockCreateEinsatz = {
        mutateAsync: mockMutateAsync,
        isPending: false,
        error: null,
    };

    // Test-Utilities
    const createQueryClient = () => new QueryClient({
        defaultOptions: {
            queries: { retry: false },
            mutations: { retry: false },
        },
    });

    const renderComponent = (queryClient = createQueryClient()) => {
        return render(
            <QueryClientProvider client={queryClient}>
                <MemoryRouter>
                    <CreateInitialEinsatz />
                </MemoryRouter>
            </QueryClientProvider>
        );
    };

    beforeEach(() => {
        vi.clearAllMocks();
        (useCreateEinsatz as unknown as ReturnType<typeof vi.fn>).mockReturnValue(mockCreateEinsatz);
    });

    // Unit Tests
    describe('Component Rendering', () => {
        it('sollte alle UI-Elemente korrekt rendern', () => {
            renderComponent();

            // Hauptelemente prüfen
            expect(screen.getByText('Initialen Einsatz erstellen')).toBeInTheDocument();
            expect(screen.getByText('Dev-Modus')).toBeInTheDocument();
            expect(screen.getByText('Dieser Schritt ist nur im Development-Modus sichtbar')).toBeInTheDocument();

            // Formular-Felder prüfen
            expect(screen.getByLabelText('Einsatz Name')).toBeInTheDocument();
            expect(screen.getByLabelText('Beschreibung (optional)')).toBeInTheDocument();

            // Submit-Button prüfen
            expect(screen.getByRole('button', { name: /einsatz erstellen/i })).toBeInTheDocument();
        });

        it('sollte Platzhalter-Texte korrekt anzeigen', () => {
            renderComponent();

            const nameInput = screen.getByPlaceholderText('z.B. Brandeinsatz Hauptstraße');
            const beschreibungInput = screen.getByPlaceholderText('Zusätzliche Details zum Einsatz...');

            expect(nameInput).toBeInTheDocument();
            expect(beschreibungInput).toBeInTheDocument();
        });

        it('sollte die Zeichen-Zählung für die Beschreibung anzeigen', () => {
            renderComponent();

            const beschreibungTextarea = screen.getByRole('textbox', { name: /beschreibung/i });
            expect(beschreibungTextarea).toHaveAttribute('maxlength', '500');
        });
    });

    // End-to-End-Flows
    describe('End-to-End Flows', () => {
        it('sollte den vollständigen Erstellungsflow erfolgreich durchlaufen', async () => {
            const user = userEvent.setup();
            const mockNewEinsatz = { id: '123', name: 'Test Einsatz' };
            mockMutateAsync.mockResolvedValue(mockNewEinsatz);

            renderComponent();

            // Formular ausfüllen
            await user.type(screen.getByLabelText('Einsatz Name'), 'Brandeinsatz Hauptstraße');
            await user.type(screen.getByLabelText('Beschreibung (optional)'), 'Großbrand in Mehrfamilienhaus');

            // Formular absenden
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            // Prüfen, ob die API-Funktion korrekt aufgerufen wurde
            await waitFor(() => {
                expect(mockMutateAsync).toHaveBeenCalledWith({
                    name: 'Brandeinsatz Hauptstraße',
                    beschreibung: 'Großbrand in Mehrfamilienhaus',
                });
            });

            // Prüfen, ob zur Einsätze-Übersicht navigiert wird
            await waitFor(() => {
                expect(mockNavigate).toHaveBeenCalledWith('/app/einsaetze');
            });
        });

        it('sollte den Erstellungsflow ohne Beschreibung erfolgreich durchlaufen', async () => {
            const user = userEvent.setup();
            const mockNewEinsatz = { id: '123', name: 'Test Einsatz' };
            mockMutateAsync.mockResolvedValue(mockNewEinsatz);

            renderComponent();

            // Nur den Namen ausfüllen
            await user.type(screen.getByLabelText('Einsatz Name'), 'Einsatz ohne Beschreibung');

            // Formular absenden
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            // Prüfen, ob die API-Funktion korrekt aufgerufen wurde (ohne beschreibung)
            await waitFor(() => {
                expect(mockMutateAsync).toHaveBeenCalledWith({
                    name: 'Einsatz ohne Beschreibung',
                    beschreibung: undefined,
                });
            });
        });

        it('sollte Fehlbehandlung korrekt durchführen', async () => {
            const user = userEvent.setup();
            const errorMessage = 'Netzwerkfehler beim Erstellen';
            mockMutateAsync.mockRejectedValue(new Error(errorMessage));

            renderComponent();

            // Formular ausfüllen
            await user.type(screen.getByLabelText('Einsatz Name'), 'Test Einsatz');

            // Formular absenden
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            // Prüfen, ob Fehlermeldung angezeigt wird
            await waitFor(() => {
                expect(screen.getByText(errorMessage)).toBeInTheDocument();
            });

            // Navigation sollte nicht stattfinden
            expect(mockNavigate).not.toHaveBeenCalled();
        });
    });

    // Integrationstests
    describe('Form Integration', () => {
        it('sollte Validierungsfehler für leeren Namen anzeigen', async () => {
            const user = userEvent.setup();
            renderComponent();

            // Formular ohne Name absenden
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            // Prüfen, ob Validierungsfehler angezeigt wird
            await waitFor(() => {
                expect(screen.getByText('Bitte geben Sie einen Namen für den Einsatz ein')).toBeInTheDocument();
            });

            // API sollte nicht aufgerufen werden
            expect(mockMutateAsync).not.toHaveBeenCalled();
        });

        it('sollte Validierungsfehler für zu kurzen Namen anzeigen', async () => {
            const user = userEvent.setup();
            renderComponent();

            // Sehr kurzen Namen eingeben
            await user.type(screen.getByLabelText('Einsatz Name'), 'A');
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            // Prüfen, ob Validierungsfehler angezeigt wird
            await waitFor(() => {
                expect(screen.getByText('Der Name muss mindestens 2 Zeichen lang sein')).toBeInTheDocument();
            });
        });

        it('sollte Leerzeichen am Anfang und Ende entfernen', async () => {
            const user = userEvent.setup();
            mockMutateAsync.mockResolvedValue({ id: '123', name: 'Test' });

            renderComponent();

            // Text mit Leerzeichen eingeben
            await user.type(screen.getByLabelText('Einsatz Name'), '  Test Einsatz  ');
            await user.type(screen.getByLabelText('Beschreibung (optional)'), '  Test Beschreibung  ');

            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            // Prüfen, ob Leerzeichen entfernt wurden
            await waitFor(() => {
                expect(mockMutateAsync).toHaveBeenCalledWith({
                    name: 'Test Einsatz',
                    beschreibung: 'Test Beschreibung',
                });
            });
        });

        it('sollte leere Beschreibung als undefined behandeln', async () => {
            const user = userEvent.setup();
            mockMutateAsync.mockResolvedValue({ id: '123', name: 'Test' });

            renderComponent();

            await user.type(screen.getByLabelText('Einsatz Name'), 'Test Einsatz');
            await user.type(screen.getByLabelText('Beschreibung (optional)'), '   '); // Nur Leerzeichen

            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            await waitFor(() => {
                expect(mockMutateAsync).toHaveBeenCalledWith({
                    name: 'Test Einsatz',
                    beschreibung: undefined,
                });
            });
        });
    });

    // State-Management-Tests
    describe('State Management', () => {
        it('sollte Loading-Zustand korrekt anzeigen', () => {
            (useCreateEinsatz as unknown as ReturnType<typeof vi.fn>).mockReturnValue({
                ...mockCreateEinsatz,
                isPending: true,
            });

            renderComponent();

            const submitButton = screen.getByRole('button', { name: /wird erstellt/i });
            expect(submitButton).toBeInTheDocument();
            expect(submitButton).toHaveClass('ant-btn-loading');
        });

        it('sollte Fehlerzustand in der UI anzeigen', async () => {
            const user = userEvent.setup();
            renderComponent();

            // Manuell einen Fehler setzen
            await user.type(screen.getByLabelText('Einsatz Name'), 'Test');
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            // Fehler simulieren
            mockMutateAsync.mockRejectedValue(new Error('Test Fehler'));

            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            await waitFor(() => {
                expect(screen.getByText('Test Fehler')).toBeInTheDocument();
            });
        });

        it('sollte Formular-State nach Fehler beibehalten', async () => {
            const user = userEvent.setup();
            mockMutateAsync.mockRejectedValue(new Error('Fehler'));

            renderComponent();

            const nameInput = screen.getByLabelText('Einsatz Name');
            await user.type(nameInput, 'Test Einsatz');
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            await waitFor(() => {
                expect(screen.getByText('Fehler', { selector: '.ant-alert-error .ant-alert-description' })).toBeInTheDocument();
            });

            // Formular-Werte sollten erhalten bleiben
            expect(nameInput).toHaveValue('Test Einsatz');
        });
    });

    // Snapshot-Tests
    describe('Snapshots', () => {
        it('sollte einen Snapshot im Standard-Zustand erstellen', () => {
            const { container } = renderComponent();
            expect(container).toMatchSnapshot();
        });

        it('sollte einen Snapshot im Loading-Zustand erstellen', () => {
            (useCreateEinsatz as unknown as ReturnType<typeof vi.fn>).mockReturnValue({
                ...mockCreateEinsatz,
                isPending: true,
            });

            const { container } = renderComponent();
            expect(container).toMatchSnapshot();
        });

        it('sollte einen Snapshot mit Fehlermeldung erstellen', async () => {
            const user = userEvent.setup();
            renderComponent();

            // Fehler provozieren
            await user.type(screen.getByLabelText('Einsatz Name'), 'Test');
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            mockMutateAsync.mockRejectedValue(new Error('Test Fehler'));
            await user.click(screen.getByRole('button', { name: /einsatz erstellen/i }));

            await waitFor(() => {
                expect(screen.getByText('Test Fehler')).toBeInTheDocument();
            });

            const { container } = renderComponent();
            expect(container).toMatchSnapshot();
        });
    });

    // Accessibility Tests
    describe('Accessibility', () => {
        it('sollte korrekte ARIA-Labels haben', () => {
            renderComponent();

            expect(screen.getByRole('textbox', { name: /einsatz name/i })).toBeInTheDocument();
            expect(screen.getByRole('textbox', { name: /beschreibung/i })).toBeInTheDocument();
            expect(screen.getByRole('button', { name: /einsatz erstellen/i })).toBeInTheDocument();
        });

        it('sollte required-Attribute korrekt setzen', () => {
            renderComponent();

            const nameInput = screen.getByLabelText('Einsatz Name');
            const beschreibungInput = screen.getByLabelText('Beschreibung (optional)');

            expect(nameInput).toBeRequired();
            expect(beschreibungInput).not.toBeRequired();
        });
    });
}); 