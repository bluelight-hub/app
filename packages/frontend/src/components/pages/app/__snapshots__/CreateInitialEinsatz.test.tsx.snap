// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreateInitialEinsatz > Snapshots > sollte einen Snapshot im Loading-Zustand erstellen 1`] = `
<div>
  <div
    class="flex items-center justify-center min-h-screen p-4"
  >
    <div
      class="ant-card ant-card-bordered w-full max-w-md shadow-lg css-dev-only-do-not-override-1vjf2v5"
    >
      <div
        class="ant-card-head"
        style="text-align: center; font-size: 1.5rem;"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Initialen Einsatz erstellen
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-alert ant-alert-info ant-alert-with-description mb-4 css-dev-only-do-not-override-1vjf2v5"
          data-show="true"
          role="alert"
        >
          <span
            aria-label="info-circle"
            class="anticon anticon-info-circle ant-alert-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="info-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
              />
            </svg>
          </span>
          <div
            class="ant-alert-content"
          >
            <div
              class="ant-alert-message"
            >
              Dev-Modus
            </div>
            <div
              class="ant-alert-description"
            >
              Dieser Schritt ist nur im Development-Modus sichtbar
            </div>
          </div>
        </div>
        <form
          class="ant-form ant-form-vertical ant-form-large css-dev-only-do-not-override-1vjf2v5"
        >
          <div
            class="ant-form-item css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1vjf2v5"
              >
                <label
                  class="ant-form-item-required"
                  for="name"
                  title="Einsatz Name"
                >
                  Einsatz Name
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-lg css-dev-only-do-not-override-1vjf2v5 ant-input-outlined"
                      id="name"
                      maxlength="100"
                      placeholder="z.B. Brandeinsatz Hauptstraße"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1vjf2v5"
              >
                <label
                  class=""
                  for="beschreibung"
                  title="Beschreibung (optional)"
                >
                  Beschreibung (optional)
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper ant-input-textarea-affix-wrapper ant-input-affix-wrapper-lg ant-input-textarea-show-count css-dev-only-do-not-override-1vjf2v5 ant-input-show-count ant-input-outlined"
                      data-count="0 / 500"
                    >
                      <textarea
                        class="ant-input ant-input-lg css-dev-only-do-not-override-1vjf2v5"
                        id="beschreibung"
                        maxlength="500"
                        placeholder="Zusätzliche Details zum Einsatz..."
                        rows="3"
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span
                          class="ant-input-data-count"
                        >
                          0 / 500
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item mb-0 css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-1vjf2v5 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-lg ant-btn-loading w-full"
                      type="submit"
                    >
                      <span
                        class="ant-btn-icon ant-btn-loading-icon"
                      >
                        <span
                          aria-label="loading"
                          class="anticon anticon-loading anticon-spin"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="loading"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 1024 1024"
                            width="1em"
                          >
                            <path
                              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                            />
                          </svg>
                        </span>
                      </span>
                      <span>
                        Wird erstellt...
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`CreateInitialEinsatz > Snapshots > sollte einen Snapshot im Standard-Zustand erstellen 1`] = `
<div>
  <div
    class="flex items-center justify-center min-h-screen p-4"
  >
    <div
      class="ant-card ant-card-bordered w-full max-w-md shadow-lg css-dev-only-do-not-override-1vjf2v5"
    >
      <div
        class="ant-card-head"
        style="text-align: center; font-size: 1.5rem;"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Initialen Einsatz erstellen
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-alert ant-alert-info ant-alert-with-description mb-4 css-dev-only-do-not-override-1vjf2v5"
          data-show="true"
          role="alert"
        >
          <span
            aria-label="info-circle"
            class="anticon anticon-info-circle ant-alert-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="info-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
              />
            </svg>
          </span>
          <div
            class="ant-alert-content"
          >
            <div
              class="ant-alert-message"
            >
              Dev-Modus
            </div>
            <div
              class="ant-alert-description"
            >
              Dieser Schritt ist nur im Development-Modus sichtbar
            </div>
          </div>
        </div>
        <form
          class="ant-form ant-form-vertical ant-form-large css-dev-only-do-not-override-1vjf2v5"
        >
          <div
            class="ant-form-item css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1vjf2v5"
              >
                <label
                  class="ant-form-item-required"
                  for="name"
                  title="Einsatz Name"
                >
                  Einsatz Name
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-lg css-dev-only-do-not-override-1vjf2v5 ant-input-outlined"
                      id="name"
                      maxlength="100"
                      placeholder="z.B. Brandeinsatz Hauptstraße"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1vjf2v5"
              >
                <label
                  class=""
                  for="beschreibung"
                  title="Beschreibung (optional)"
                >
                  Beschreibung (optional)
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper ant-input-textarea-affix-wrapper ant-input-affix-wrapper-lg ant-input-textarea-show-count css-dev-only-do-not-override-1vjf2v5 ant-input-show-count ant-input-outlined"
                      data-count="0 / 500"
                    >
                      <textarea
                        class="ant-input ant-input-lg css-dev-only-do-not-override-1vjf2v5"
                        id="beschreibung"
                        maxlength="500"
                        placeholder="Zusätzliche Details zum Einsatz..."
                        rows="3"
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span
                          class="ant-input-data-count"
                        >
                          0 / 500
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item mb-0 css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-1vjf2v5 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-lg w-full"
                      type="submit"
                    >
                      <span
                        class="ant-btn-icon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 256 256"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
                          />
                        </svg>
                      </span>
                      <span>
                        Einsatz erstellen
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`CreateInitialEinsatz > Snapshots > sollte einen Snapshot mit Fehlermeldung erstellen 1`] = `
<div>
  <div
    class="flex items-center justify-center min-h-screen p-4"
  >
    <div
      class="ant-card ant-card-bordered w-full max-w-md shadow-lg css-dev-only-do-not-override-1vjf2v5"
    >
      <div
        class="ant-card-head"
        style="text-align: center; font-size: 1.5rem;"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Initialen Einsatz erstellen
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-alert ant-alert-info ant-alert-with-description mb-4 css-dev-only-do-not-override-1vjf2v5"
          data-show="true"
          role="alert"
        >
          <span
            aria-label="info-circle"
            class="anticon anticon-info-circle ant-alert-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="info-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
              />
            </svg>
          </span>
          <div
            class="ant-alert-content"
          >
            <div
              class="ant-alert-message"
            >
              Dev-Modus
            </div>
            <div
              class="ant-alert-description"
            >
              Dieser Schritt ist nur im Development-Modus sichtbar
            </div>
          </div>
        </div>
        <form
          class="ant-form ant-form-vertical ant-form-large css-dev-only-do-not-override-1vjf2v5"
        >
          <div
            class="ant-form-item css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1vjf2v5"
              >
                <label
                  class="ant-form-item-required"
                  for="name"
                  title="Einsatz Name"
                >
                  Einsatz Name
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-lg css-dev-only-do-not-override-1vjf2v5 ant-input-outlined"
                      id="name"
                      maxlength="100"
                      placeholder="z.B. Brandeinsatz Hauptstraße"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1vjf2v5"
              >
                <label
                  class=""
                  for="beschreibung"
                  title="Beschreibung (optional)"
                >
                  Beschreibung (optional)
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper ant-input-textarea-affix-wrapper ant-input-affix-wrapper-lg ant-input-textarea-show-count css-dev-only-do-not-override-1vjf2v5 ant-input-show-count ant-input-outlined"
                      data-count="0 / 500"
                    >
                      <textarea
                        class="ant-input ant-input-lg css-dev-only-do-not-override-1vjf2v5"
                        id="beschreibung"
                        maxlength="500"
                        placeholder="Zusätzliche Details zum Einsatz..."
                        rows="3"
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span
                          class="ant-input-data-count"
                        >
                          0 / 500
                        </span>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item mb-0 css-dev-only-do-not-override-1vjf2v5"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1vjf2v5"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1vjf2v5"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-1vjf2v5 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-lg w-full"
                      type="submit"
                    >
                      <span
                        class="ant-btn-icon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 256 256"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
                          />
                        </svg>
                      </span>
                      <span>
                        Einsatz erstellen
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
